---
title: Boot Camps
---

Current state of Chado Natural Diversity module
-----------------------------------------------

*People interested in attending:* [Vladimir](User:Vg34 "wikilink"),
[Matt](User:diapriid@gmail.com "wikilink"),
[Jim](User:balhoff@nescent.org "wikilink"),
[Maren](User:friesen@usc.edu "wikilink"),
[Yuri](User:ybendana@usc.edu "wikilink"),
[Richard](User:rdhayes@lbl.gov "wikilink") (add your name)

Relevant resources and questions:

-   [Chado Natural Diversity Module Working
    Group](http://gmod.org/wiki/Chado_Natural_Diversity_Module_Working_Group)
-   [the talk
    page](http://gmod.org/wiki/Talk:Chado_Natural_Diversity_Module_Working_Group)
    for the above, which actually appears more informative!
-   Has the new Chadoesque version of the Chado ND schema mentioned on
    these pages already been released? Where could we see it?
-   The [Diversity and Phenotypes](Diversity_and_Phenotypes "wikilink")
    abstract mentions "A recently reported prototype of a Chado data
    adapter for Phenote, GMODs phenotype annotation tool, could be
    generalized to become the data persistence interface for such data."
    Is there a paper or another source that reports on this adapter?

*I would add explaining the current phenotype implementation. [Dave
C](User:Dpc13 "wikilink")*

[Chado](gmod:Chado "wikilink") essentials
-----------------------------------------

*People interested in attending:*
[Matt](User:diapriid@gmail.com "wikilink"),
[Vladimir](User:Vg34 "wikilink"),
[Chris](User:cjfields@illinois.edu "wikilink"),
[Jim](User:balhoff@nescent.org "wikilink"),
[Yuri](User:ybendana@usc.edu "wikilink"),
[Duke](User:jonathan@leto.net "wikilink"),
[Richard](User:rdhayes@lbl.gov "wikilink") (add your name)

The Chado Way, Features, CVs, DBXREFs and IDs, ..

Mechanics of using Chado: installing, loading data.
--[Vladimir](User:Vg34 "wikilink")

Genome Browsing Essentials
--------------------------

*People interested in attending:*
[Matt](User:diapriid@gmail.com "wikilink"),
[Duke](User:jonathan@leto.net "wikilink"),
[Richard](User:rdhayes@lbl.gov "wikilink"),
[Greg](User:greg@ebi.ac.uk "wikilink") (add your name)

[GBrowse](gmod:GBrowse "wikilink"), [JBrowse](gmod:JBrowse "wikilink"),
[GBrowse\_syn](gmod:GBrowse_syn "wikilink"),
[CoGe](http://genomevolution.org), Mauve.;

From [*Nature Methods* Supplement on visualizing biological
data](http://www.nature.com/nmeth/journal/v7/n3s/index.html), March
2010:

-   [Visualizing genomes: techniques and
    challenges](http://www.nature.com/nmeth/journal/v7/n3s/abs/nmeth.1422.html),
    CB Nielsen, *et al.*

This issue also has:

-   [Visualization of multiple alignments, phylogenies and gene family
    evolution](http://www.nature.com/nmeth/journal/v7/n3s/abs/nmeth.1434.html),
    JB Proctor, *et al.*

Comparative Genomics Visualization and Interactive Tools
--------------------------------------------------------

*People interested in attending:*
[Duke](User:jonathan@leto.net "wikilink"),
[Richard](User:rdhayes@lbl.gov "wikilink"),
[Greg](User:greg@ebi.ac.uk "wikilink")

-   Ben Hitz: [SGD](http://www.yeastgenome.org/)
    -   Tentative: Outline of the 1000 closely related genome problem
-   Eric Lyons: [CoGe](http://genomevolution.org/)
    -   Pair-wise whole genome syntenic analyses, advanced visualization
        using synonymous mutation rates, high-resolution sequence
        analyses using multiple genomic regions
    -   Will focus on a plant genome comparisons where multiple shared
        and independent whole genome duplications creates multiple
        syntenic regions (mappings) for any given genomic region.

