---
title: Main Page
---

<center>
|                                                                     |                                                  |
|---------------------------------------------------------------------|--------------------------------------------------|
| <div style="text-align: right; font-size: 200%; line-height: 120%"> 
 <b>GMOD Tools for                                                    
 Evolutionary Biology Hackathon</b>                                   
 November 8-12, 2010                                                  
 [NESCent](http://nescent.org/), Durham, North Carolina, USA          
                                                                      
 </div>                                                               | ![](EvoHackathonLogo.png "EvoHackathonLogo.png") |

</center>
GMOD will be holding a hackathon November 8-12, 2010, at the [National
Evolutionary Synthesis Center (NESCent)](http://nescent.org) in Durham,
North Carolina. This hackathon will focus on improving GMOD's support
for [evolutionary biology](gmod:Category:Evolution "wikilink").

The [Open Call for
Participation](gmod:GMOD_Evo_Hackathon_Open_Call "wikilink") went out on
August 1, 2010, and remained open until August 25. Participants have
been selected and notified of their status.

Synopsis
========

This hackathon will fill critical gaps in the capabilities of the
Generic Model Organism Database (GMOD) toolbox that currently limit its
utility for evolutionary research. Specifically, we will [focus
on](#Specific_Objectives "wikilink") tools for 1) viewing comparative
genomics data; 2) visualizing phylogenomic data; and 3) supporting
population diversity data and phenotype annotation.

The event will bring together a group of about 20 software developers,
end-user representatives, and documentation experts who would otherwise
not meet. The participants will include key developers of GMOD
components that currently lack features critical for emerging
evolutionary biology research, developers of informatics tools in
evolutionary research that lack GMOD integration, and informatics-savvy
biologists who can represent end-user requirements.

This hackathon will provide a unique opportunity to infuse the community
of GMOD developers with a heightened awareness of unmet needs in
evolutionary biology that GMOD components have the potential to fill,
and for tool developers in evolutionary biology to better understand how
best to extend or integrate with already existing GMOD components.

Background
==========

The GMOD project is a confederation of intercompatible open-source
projects developing software tools for storing, managing, curating, and
publishing biological data. Although the GMOD project originated from
the goal of developing a generic tool set for common needs among model
organism databases, GMOD tools are meanwhile used by many large and
small, collaborative and single-investigator biological database
projects for the dissemination of experimental results and curated
knowledge.

GMOD's software tools provide a powerful and feature-rich basis for
working with biological, in particular genomic and other molecular data.
However, due to GMOD's historical emphasis on single-genome projects
many GMOD tools still lack features that are critical to effectively
support the comparative, phylogenetic, and natural diversity-oriented
questions frequently asked in evolutionary research.

Recent developments have given rise to a window of opportunity for
forging collaborations towards filling this gap. In particular, the cost
of collecting comparative molecular data on a large or even genomic
scale has recently dropped dramatically, primarily thanks to
next-generation high-throughput sequencing technologies. This has
enabled evolutionary researchers to bring genome-scale molecular data to
bear on key evolutionary questions. It has also allowed single
organism-focused molecular biology labs, who represent GMOD's
traditional user base, to broaden out to multi-organism comparative
approaches. Bringing these two communities with increasingly shared
interests and complementary scientific and technical expertise together
offers an opportunity to start filling GMOD's gaps in these areas while
building on its existing strengths. In addition, such direct interaction
will heighten future awareness of needs of evolutionary researchers
among GMOD developers who have so far mostly supported its traditional
user base, and can in the long term increase the ranks of GMOD
contributors from a field it was not originally designed to serve.

The hackathon format is ideally suited to realize this opportunity. Its
strengths lie in facilitating face-to-face interaction among people with
complementary expertise, and collaborative work on tangible products
that can form the basis of continued partnerships long beyond the end of
the meeting.

Specific Objectives
===================

Organizers have identified the following broad themes for focusing work
at the event. Before and at the hackathon, the participants will refine
and distill these and other options into concrete implementation
targets.

-   [Viewing tools for comparative genomics
    data](Comparative_Genomics_Visualization "wikilink")
-   [Visualization of phylogenetic data and
    trees](Phylogenetics_and_Tree_Visualization "wikilink")
-   [Population Diversity and Phenotype
    support](Diversity_and_Phenotypes "wikilink")

Hackathon Structure
===================

Before the Event
----------------

Discussion of ideas and sometimes even design actually starts well
before the hackathon, on mailing lists, wiki pages, and conference calls
set up among accepted attendees. This advance work lays the foundation
for participants to be productive from the very first day. This also
means that participants should be willing to contribute some time in
advance of the hackathon itself to participate in this preparatory
discussion.

During the Event
----------------

Typically, hackathon participants use the morning of the first day of
the event to organize themselves into working groups of between 3 and 6
people, each with a focused implementation objective. Ideas and
objectives are discussed, and attendees coalesce around the projects in
which they have the most experience or interest.

Deliverables / Event Results
----------------------------

The meetingâ€™s attendance, working groups, and outcomes will be fully
logged and documented on the [GMOD wiki](gmod:Main_Page "wikilink").
Each working group during the event will typically have its own wiki
page, linked from the main EvoHack page, where it documents its minutes
and design notes, and provides links to the code and documentation it
produces. Also, since GMOD and NESCent are both committed to open source
principles, all code and documentation produced by participants during
the hackathon must be published under an [OSI-approved open source
license](http://www.opensource.org/licenses/category). As contributions
to existing GMOD tools, all hackathon products will most likely satisfy
this requirement automatically.

Sponsorship
===========

|                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |                                              |
|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------|
| This event is sponsored by the [US National Evolutionary Synthesis Center (NESCent)](http://www.nescent.org/) through its [Informatics Whitepapers program](http://www.nescent.org/informatics/whitepapers.php). NESCent [promotes the synthesis of information, concepts and knowledge](http://www.nescent.org/science/proposals.php) to address significant, emerging, or novel questions in evolutionary science and its applications. NESCent achieves this by supporting research and education across disciplinary, institutional, geographic, and demographic boundaries. | ![](NESCentLogo152.jpg "NESCentLogo152.jpg") |

MediaWiki Help if you need it
=============================

-   [User's
    Guide](http://meta.wikipedia.org/wiki/MediaWiki_User%27s_Guide)
-   [MediaWiki FAQ](http://www.mediawiki.org/wiki/Help:FAQ)

